src/main/webapp/app/entities/lecture.model.ts

```diff
@@ -21,6 +22,31 @@ export class Lecture implements BaseEntity {
     channelName?: string;
     isAtLeastEditor?: boolean;
     isAtLeastInstructor?: boolean;
+    ingested?: IngestionState;
 
-    constructor() {}
+    constructor() {
+        this.ingested = this.checkIngestionState ? this.checkIngestionState() : undefined;
+    }
+
+    private checkIngestionState?(): IngestionState {
+        if (!this.lectureUnits) {
+            return IngestionState.NOT_STARTED;
+        }
+
+        const attachmentUnits = this.lectureUnits.filter((unit) => unit.type === 'attachment') as AttachmentUnit[];
+        const allDone = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.DONE);
+        const allNotStarted = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.NOT_STARTED);
+        const allFailed = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.ERROR);
+
+        if (allDone) return IngestionState.DONE;
+        if (allFailed) return IngestionState.ERROR;
+        if (allNotStarted) return IngestionState.NOT_STARTED;
+
+        return IngestionState.PARTIALLY_INGESTED;
```

Comment:
Should differentiate between the cases where some attachment units are already ingested and
1. some other are not yet ingested
2. some other had errors

Reply:
This can be seen when the lecture is accessed, the icons show what state each lecture unit has.